File: 2271-RearrangeArrayElementsBySign/2271-RearrangeArrayElementsBySign.py
# Last updated: 1/15/2026, 9:29:38 PM
class Solution(object):
    def rearrangeArray(self, nums):
        n = len(nums)
        res = [0] * n
        
        posindex = 0
        negindex = 1
        
        for i in nums:
            if i > 0:
                res[posindex] = i
                posindex += 2
            else:
                res[negindex] = i
                negindex += 2
                
        return res

File: 2271-RearrangeArrayElementsBySign/2271-RearrangeArrayElementsBySign.c
// Last updated: 1/15/2026, 9:29:36 PM
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* rearrangeArray(int* nums, int numsSize, int* returnSize) {
    *returnSize = numsSize;

    int *res = malloc(numsSize * sizeof(int));

    int posindex = 0;
    int negindex = 1;

    int i = 0;
    while (i < numsSize) {
        if (nums[i] > 0) {
            res[posindex] = nums[i];
            posindex += 2;
        } else {
            res[negindex] = nums[i];
            negindex += 2;
        }
        i++;
    }

    return res;
}